tests/robot-tests/tests/admin/bau/create_data_block_with_chart.robot

```diff
@@ -335,11 +333,34 @@ Configure basic line chart
     user enters text into element    id:chartConfigurationForm-height    400
     user enters text into element    id:chartConfigurationForm-width    900
 
+Validate changing data sets
     user clicks link    Data sets
     user waits until h3 is visible    Data sets
+
+    user chooses select option    id:chartDataSetsConfigurationForm-location    Nailsea Youngwood
+    user clicks button    Add data set
+
+    user chooses select option    id:chartDataSetsConfigurationForm-location    Syon
+    user clicks button    Add data set
+
+    user chooses select option    id:chartDataSetsConfigurationForm-location    Bolton 001
+    user clicks button    Add data set
+
+    user checks chart legend item contains    id:chartBuilderPreview    1    Admission Numbers (Nailsea Youngwood)
+    user checks chart legend item contains    id:chartBuilderPreview    2    Admission Numbers (Syon)
+    user checks chart legend item contains    id:chartBuilderPreview    3    Admission Numbers (Bolton 001)
+
+    user clicks button    Remove all
+    user clicks button    Confirm
+
+    user checks page contains    Configure the chart and add data to view a preview
```

Comment:
Would go a little further here and check that the data sets table has been emptied too.